### ISSUE TITLE
Unable to use python package

### ISSUE THREAD

**Issue by christophgil** [Nesvilab/FragPipe#978]
Using Fragpipe 19.1

I psecified python 
/opt/conda/envs/py39/bin/python

However it is not finding easypqp
I unset PYTHONPATH and still not finding easypqp

**Checking installed easypqp**
 /opt/conda/envs/py39/bin/pip show easypqp
Name: easypqp
Version: 0.1.35
Summary: EasyPQP: Simple library generation for OpenSWATH
....
Location: /opt/conda/envs/py39/lib/python3.9/site-packages

**Workaround:**
Explicitely setting PYTHONPATH to /opt/conda/envs/py39/lib/python3.9/site-packages
solves this issue.

**Comment by fcyu** [comment]
I am sorry, but is there anything what we should fix or improve?

Thanks,

Fengchao

**Comment by christophgil** [comment]
with
You have the path of the python binary.
pip show easypqp
you can detect the location of the local installation and then set
PYTHONPATH in the script accordingly.

**Comment by guoci** [comment]
@christophgil Can you enter in your terminal
`/opt/conda/envs/py39/bin/python -m pip show --files easypqp`
and show me the output?

**Comment by christophgil** [comment]
This is what is needed in my case:
 export PYTHONPATH=/opt/conda/envs/py39/lib/python3.9/site-packages


/opt/conda/envs/py39/bin/python -m pip show --files easypqp
Name: easypqp
Version: 0.1.35
Summary: EasyPQP: Simple library generation for OpenSWATH
Home-page: https://github.com/grosenberger/easypqp
Author: George Rosenberger
Author-email: ***@***.***
License:
Location: /opt/conda/envs/py39/lib/python3.9/site-packages
Requires: biopython, Click, matplotlib, numpy, pandas, pyopenms,
scikit-learn, scipy, seaborn, statsmodels, tqdm
Required-by:
Files:
  ../../../bin/easypqp
  easypqp-0.1.35.dist-info/INSTALLER
  easypqp-0.1.35.dist-info/LICENSE
  easypqp-0.1.35.dist-info/METADATA
  easypqp-0.1.35.dist-info/RECORD
  easypqp-0.1.35.dist-info/REQUESTED
  easypqp-0.1.35.dist-info/WHEEL
  easypqp-0.1.35.dist-info/entry_points.txt
  easypqp-0.1.35.dist-info/top_level.txt
  easypqp/__init__.py
  easypqp/__pycache__/__init__.cpython-39.pyc
  easypqp/__pycache__/convert.cpython-39.pyc
  easypqp/__pycache__/library.cpython-39.pyc
  easypqp/__pycache__/main.cpython-39.pyc
  easypqp/__pycache__/test_convert.cpython-39.pyc
  easypqp/__pycache__/unimoddb.cpython-39.pyc
  easypqp/convert.py
  easypqp/data/unimod.xml
  easypqp/library.py
  easypqp/main.py
  easypqp/test_convert.py
  easypqp/unimoddb.py

**Comment by guoci** [comment]
@christophgil What happens if you unset `PYTHONPATH`? What is the output then?

**Comment by christophgil** [comment]
Works fine from command line.
But in the Fragpipe GUI, easypqp is not recognized.
I also tried export PYTHONPATH=''

With the Runtime.exec java function, you can pass env variables.

I suggest that you first run the pip show command from java to determine
the location and then
run python with Runtime.exec with the env variable specified.



Here the output:

unset PYTHONPATH  # No matter whether unset or export to any nonsense
string. Always works
/opt/conda/envs/py39/bin/python -m pip show --files easypqp
Name: easypqp
Version: 0.1.35
Summary: EasyPQP: Simple library generation for OpenSWATH
Home-page: https://github.com/grosenberger/easypqp
Author: George Rosenberger
Author-email: ***@***.***
License:
Location: /opt/conda/envs/py39/lib/python3.9/site-packages
Requires: biopython, Click, matplotlib, numpy, pandas, pyopenms,
scikit-learn, scipy, seaborn, statsmodels, tqdm
Required-by:
Files:
  ../../../bin/easypqp

**Comment by guoci** [comment]
@christophgil you are using a conda env and the commands you run needs `conda activate`.
The problem is when you run FragPipe you did not do `conda activate` before starting it.

**Comment by christophgil** [comment]
Thanks!

Another issue I had was to use  a global FragPipe installation for
different users such that each user can have its own settings and
configuration.
I was expecting that the local config files maybe in ~/.fragpipe/  take
precedence over the global  fragpipe-runtime.cache and   fragpipe-ui.cache

Is this supposed to work or should each user have it's own installation?


I solved this by embeding the global installation in an overlay FS
(unionfs-fuse) so it works for us.

Best

Christoph

**Comment by fcyu** [comment]
Hi Christoph,

Can you try if configuring the temp file directory works for you: https://github.com/Nesvilab/FragPipe/issues/755#issuecomment-1180889147 ?

Thanks,

Fengchao

**Comment by christophgil** [comment]
Hi Fengchao,

Please let me ask a question on file names.
we observed that  Fragpipe does not work on bruker file folders that have a
dot in the name.
Therefore we need to rename files like 20230102_abcd_0.5ug_defg.d into
20230102_abcd_0_5ug_defg.d.
Could dots in the name be supported?

Best
Christoph

**Comment by fcyu** [comment]
Hi Christoph,

Some tools, such as PeptideProphet if I remember it correctly, have some issues when there is a dot in the file name. That is why FragPipe wants you to rename the file. 

But, you can skip the renaming step by clicking "try to rename files" and then "no":

![image](https://user-images.githubusercontent.com/6926299/216849409-816f372f-1ae8-43e3-9827-c79ac9e566db.png)

![image](https://user-images.githubusercontent.com/6926299/216849422-7e575dc7-8e2f-497b-88b3-fd4fce1e0a9c.png)

Best,

Fengchao

**Comment by christophgil** [comment]
Thank you!
Our files are already in a WORM storage.
I will try to make a virtual file system were the files are presented with
a fake name.



Best
Christoph

**Comment by christophgil** [comment]
Hi Fengchao,

I am working on a virtual FS and notice that extremely often
access  to these non existing files are requested which creates a
performance problem:
analysis.tdf-journal
analysis.tdf-wal

I guess the Bruker code is responsible is'nt it?
Is this open source code so I could inspect the code?

Best

Christoph

**Comment by fcyu** [comment]
Hi Christoph,

> I guess the Bruker code is responsible is'nt it?

Yes.

> Is this open source code so I could inspect the code?

No.

BTW, you actually no need to work on the virtual FS because you don't have to rename the file name. See my last comment: https://github.com/Nesvilab/FragPipe/issues/978#issuecomment-1418281841

Best,

Fengchao

**Comment by christophgil** [comment]
Cool!

Nevertheless, we have zip files in a WORM
Currently we mount, the zips,but this is awkward.
Best Christoph
